### PR TITLE
Feature standardized testing

### PR DESCRIPTION
**Pull request by rrudolfo**
Has 4 test classes that include: RelationshipTest, UMLClassTest, UMLFieldsTest, and UMLMethodsTest.

### CODE REVIEW

File: UML editor/src/test/java/UMLMethodsTests.java [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+        umlMethods.addParameter("Test", "AddDog", "String");
+        UMLClassInfo classInfo = UMLClass.classMap.get("Test");
+        UMLMethodInfo methodInfo = classInfo.getMethodByName("AddDog");
+        assertTrue(methodInfo.getParameters().contains("String"));
+    }
+
+    /**
+     * Tests removing a parameter from an existing method within a class.
+     */
+    @Test
```
Comment (AaronWB) [inline UML editor/src/test/java/UMLMethodsTests.java:167]: Small typo where int is added to the parameter list but String is expected and "removed", test passes since String was never in the array to begin with.

Needs fixed

**Review by AaronWB — CHANGES_REQUESTED** [review]
Small typo mistake needs to be changed but otherwise these tests are a good baseline to make sure all methods do their expected outcomes without more advanced logic.

File: UML editor/src/test/java/UMLMethodsTests.java (hunk 1 above)
Comment (rrudolfo) [reply UML editor/src/test/java/UMLMethodsTests.java:167]: Thank you for the review, I have fixed that typo.

**Review by AaronWB — APPROVED** [review]
Typo is fixed, code looks all good.

File: UML editor/src/test/java/RelationshipTests.java [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+        UMLClass.classMap.put("CAT", new UMLClassInfo("CAT"));
+
+        Relationship.addRelationship("DOG", "BIRD");
+        Relationship.addRelationship("DOG", "CAT");
+        Relationship.removeAttachedRelationships("DOG");
+
+        assertFalse(Relationship.relationshipExists("DOG", "BIRD"));
+        assertFalse(Relationship.relationshipExists("DOG", "CAT"));
+        assertEquals(0, Relationship.relationshipList.size());
+    }
```
Comment (colbycrutcher) [inline UML editor/src/test/java/RelationshipTests.java:104]: Looks good

File: UML editor/src/test/java/UMLClassTests.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+     * Tests renaming an existing class in the class map.
+     */
+    @Test
+    public void testRenamingClass() {
+        UMLClass.addClass("DOG");
+        UMLClass.renameClass("DOG", "CAT");
+        assertFalse(UMLClass.classMap.containsKey("DOG"));
+        assertTrue(UMLClass.classMap.containsKey("CAT"));
+    }
+
```
Comment (colbycrutcher) [inline UML editor/src/test/java/UMLClassTests.java:78]: Good edge cases

File: UML editor/src/test/java/UMLMethodsTests.java [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+    public void testRenameMethodFromNonExistingMethod() {
+        UMLClass.addClass("Test");
+        umlMethods.renameMethod("Test", "AddDog", "AddMultipleDogs");
+        UMLClassInfo classInfo = UMLClass.classMap.get("Test");
+        assertNull(classInfo.getMethodByName("AddMultipleDogs"));
+    }
+
+    /**
```
Comment (colbycrutcher) [inline UML editor/src/test/java/UMLMethodsTests.java:133]: Good bases for testing

File: UML editor/src/test/java/UMLMethodsTests.java [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.AfterEach;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * The UMLMethodsTests class contains test cases to validate the functionality of UMLMethods class.
```
Comment (colbycrutcher) [inline UML editor/src/test/java/UMLMethodsTests.java:13]: Good

**Review by colbycrutcher — COMMENTED** [review]
Good base cases for testing. Thanks for writing!

**Review by colbycrutcher — APPROVED** [review]
Good bases for testing
